src/Shared/UnitTests/Vacancies.Client/Application/CommandHandlers/ApproveVacancyReviewCommandHandlerTests.cs

```diff
@@ -137,5 +140,36 @@ public async Task GivenApprovedVacancyReviewCommand_AndProviderHasBeenBlockedSin
             existingVacancy.ClosureReason.Should().Be(ClosureReason.BlockedByQa);
             _mockVacancyRepository.Verify(x => x.UpdateAsync(existingVacancy), Times.Once);
         }
+
+        [Fact]
+        public async Task GivenApprovedVacancyReviewCommand_AndProviderHasBeenBlockedSinceReviewWasCreated_ThenRaiseUpdateEmployerDashboardEvent()
+        {
+            long blockedProviderUkprn = 12345678;
+
+            var existingVacancy = _autoFixture.Build<Vacancy>()
+                                                .Without(x => x.TransferInfo)
+                                                .With(x => x.TrainingProvider, new TrainingProvider { Ukprn = blockedProviderUkprn })
+                                                .Create();
+
+            _mockVacancyRepository.Setup(x => x.GetVacancyAsync(existingVacancy.VacancyReference.Value)).ReturnsAsync(existingVacancy);
+
+            _mockVacancyReviewRepository.Setup(x => x.GetAsync(_existingReviewId)).ReturnsAsync(new VacancyReview
+            {
+                Id = _existingReviewId,
+                CreatedDate = _mockTimeProvider.Object.Now.AddHours(-5),
+                Status = ReviewStatus.UnderReview,
+                VacancyReference = existingVacancy.VacancyReference.Value,
+                VacancySnapshot = new Vacancy()
+            });
+
+            _mockBlockedOrganisationQuery.Setup(b => b.GetByOrganisationIdAsync(blockedProviderUkprn.ToString()))
+                .ReturnsAsync(new BlockedOrganisation { BlockedStatus = BlockedStatus.Blocked });
+
+            var command = new ApproveVacancyReviewCommand(_existingReviewId, "comment", new List<ManualQaFieldIndicator>(), new List<Guid>());
+
+            await _sut.Handle(command, CancellationToken.None);
+            _dashboardService.Verify(x=>x.ReBuildDashboardAsync(It.IsAny<string>()),Times.Once);
```

Comment:
Use the expected employer account id rather than `IsAny`.

Reply:
👍